src/main/java/org/marketdesignresearch/mechlib/outcomerules/ccg/constraintgeneration/UnitedConstrainedGenerator.java

```diff
@@ -18,17 +19,18 @@
 
 public class UnitedConstrainedGenerator implements ConstraintGenerator {
     private final Set<PartialConstraintGenerator> generatorAlgorithms;
-    private final Map<Good, PotentialCoalition> goodToCoalitionMap = new HashMap<>();
+    private final Map<Good, Set<PotentialCoalition>> goodToCoalitionMap = new HashMap<>();
     private final CorePaymentRule corePaymentRule;
 
     public UnitedConstrainedGenerator(BundleValueBids<?> bids, Outcome referencePoint, Set<PartialConstraintGenerator> generatorAlgorithms, CorePaymentRule corePaymentRule) {
         this.generatorAlgorithms = generatorAlgorithms;
         this.corePaymentRule = corePaymentRule;
 
         for (PotentialCoalition coalition : referencePoint.getAllocation().getPotentialCoalitions()) {
-            // TODO: assumes that there are only single-availability goods
+            // TODO: check implementation for availability of more than 1
             for (Good good : coalition.getBundle().getBundleEntries().stream().map(BundleEntry::getGood).collect(Collectors.toSet())) {
-                goodToCoalitionMap.put(good, coalition);
+            	goodToCoalitionMap.putIfAbsent(good, new HashSet<>());
+                goodToCoalitionMap.get(good).add(coalition);
```

Comment:
I studied the paper of Benedikt for my MA Basismodul, therfore I'm quite sure this solution will generalize for his SeparatabilityConstraintGeneration. Just allow more than one winning bidder per good.

Reply:
I definitely trust you on that, that's great!